### PR TITLE
Skip token in `push_to_hub`

### PR DESCRIPTION
**Pull request by qgallouedec**

### CODE REVIEW

File: trl/trainer/bco_trainer.py [hunk 1]
```diff
@@ -1432,12 +1432,12 @@ def push_to_hub(
         self,
         commit_message: Optional[str] = "End of training",
         blocking: bool = True,
-        token: Optional[str] = None,
+        # token: Optional[str] = None,  # Unlike the parent class, skip the token to mitigate security risks.
```
Comment (lewtun) [inline trl/trainer/bco_trainer.py:1435]: Can't we delete altogether instead of having this commented out code?

```suggestion
```

**Review by lewtun — COMMENTED** [review]
Thanks for the safety improvement! My main question is whether we need to have commented out args - is there a cleaner way?

File: trl/trainer/bco_trainer.py (hunk 1 above)
Comment (qgallouedec) [reply trl/trainer/bco_trainer.py]: We can. 
I just want to find a way to tell my future self not to re-add it. I'll add to the comment below, let me know if that's ok.

**Comment by qgallouedec** [comment]
Note that thanks to `@wraps(Trainer.push_to_hub)`, the docstring (```"""Overwrite the `push_to_hub` method in order to ..."""```) won't appear in the online doc. It will only be visible in the source code.

File: trl/trainer/bco_trainer.py (hunk 1 above)
Comment (lewtun) [reply trl/trainer/bco_trainer.py]: This is better, thanks!

**Review by lewtun — APPROVED** [review]
Thanks for iterating, LGTM!

**Pull request merged**